### PR TITLE
Feat: recreate release-source.yaml workflow in datavisyn/github-workflows repo

### PR DESCRIPTION
**Pull request by dv-hossam-radwan**
Closes https://github.com/datavisyn/infrastructure-k8s/issues/241

Sample output:
https://github.com/datavisyn/test-app/actions/runs/8554525622/job/23440021181
https://github.com/datavisyn/test-app/pull/39

https://github.com/datavisyn/test-app/actions/runs/8554492359/job/23439919793
https://github.com/datavisyn/test-app/pull/38

### CODE REVIEW

File: .github/workflows/release-source.yml [hunk 1]
```diff
@@ -25,283 +26,122 @@ jobs:
...
-          readme="$readme$commit_titles\n\n### Release dependencies first\n\nIn case of dependent visyn repositories follow dependency tree from the top.\n\n"
-          readme="$readme### 🏁 Finish line\n\n"
-          readme="$readme* [ ] Inform colleagues and customers about the release\n"
-          readme="$readme* [ ] Celebrate the new release 🥳"
-          echo -e "$readme"
-          echo "commits=$commit_titles"
-          echo "readme=$readme" >> "$GITHUB_OUTPUT"
-          echo "commits=$commit_titles" >> "$GITHUB_OUTPUT"
-      - name: setup git config
+          release_version="$(jq -r '.version' package.json | cut -d '-' -f 1)"
```
Comment (dvvanessastoiber) [inline .github/workflows/release-source.yml:49]: Thank you for the changes! If I see it correctly, you are currently just removing the `-SNAPSHOT` and using that version as release version. But what if we need a major or minor release? We need to handle this as well.

**Review by dvvanessastoiber — CHANGES_REQUESTED** [review]

File: .github/workflows/release-source.yml (hunk 1 above)
Comment (dv-hossam-radwan) [reply .github/workflows/release-source.yml:51]: Changes added

File: .github/workflows/release-source.yml [hunk 2]
```diff
@@ -25,283 +27,186 @@ jobs:
...
+        id: get-release-version-from-input
+        if: ${{ github.event.inputs.release_version != null }}
         run: |
-          echo "create/get release branch"
-          if git rev-parse --verify --quiet "origin/$BRANCH_NAME"; then
-            echo "branch exists!"
-            git checkout "$BRANCH_NAME"
+          input_release_version="${{ github.event.inputs.release_version }}"
+          if echo "$input_release_version" | grep -q '^[0-9.]*$'; then
+            echo "Valind input"
```
Comment (dvvanessastoiber) [inline .github/workflows/release-source.yml:59]:
```suggestion
            echo "Valid input"
```

File: .github/workflows/release-source.yml [hunk 3]
```diff
@@ -25,283 +27,186 @@ jobs:
...
           else
-            echo "create branch"
-            git checkout -b "$BRANCH_NAME"
+            echo "Error: invaled input version."
```
Comment (dvvanessastoiber) [inline .github/workflows/release-source.yml:61]:
```suggestion
            echo "Error: invalid input version."
```

File: .github/workflows/release-source.yml [hunk 4]
```diff
@@ -25,283 +27,186 @@ jobs:
...
-        uses: datavisyn/github-action-trigger-workflow@v1
-        with:
-          owner: ${{ github.repository_owner }}
-          repo: ${{ github.event.repository.name }}
-          github_token: ${{ github.token }}
-          workflow_file_name: "build.yml"
-          ref: "${{ steps.get-release-version.outputs.new_branch_name }}"
-          github_user: ${{ env.GITHUB_USER }}
-      - name: create pull request
+          git checkout -b release/"$RELEASE_VERSION"
```
Comment (dvvanessastoiber) [inline .github/workflows/release-source.yml:120]:
```suggestion
          git checkout -b release-"$RELEASE_VERSION"
```
Please avoid `/` in branch names - they sometimes cause issues in GA

File: .github/workflows/release-source.yml [hunk 5]
```diff
@@ -25,283 +27,186 @@ jobs:
...
+          # install homebrew package manager
+          yes | /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+          (echo; echo "eval \"$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)\"") >> /home/runner/.bashrc
+          eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+          sudo apt-get install build-essential
+          # install helm-docs
+          brew install norwoodj/tap/helm-docs
+          # update readme files
+          cd deploy/helm
+          helm-docs .
```
Comment (dvvanessastoiber) [inline .github/workflows/release-source.yml:189]:
```suggestion
          helm-docs --document-dependency-values
```

**Review by dvvanessastoiber — CHANGES_REQUESTED** [review]
Thank you very much for the implementation! I have a few more comments, but it looks great already! Thank you also for incorporating the special cases for helm chart repos!

File: .github/workflows/release-source.yml [hunk 6]
```diff
@@ -25,283 +27,186 @@ jobs:
...
+            if [ -e "$dir/values/values.yaml" ]; then
+                if grep -q "imageTag" "$dir/values/values.yaml"; then \
+                  sed -i "s/^imageTag.*/imageTag: \&imageTag \"v$RELEASE_VERSION\"/" "$dir"/values/values.yaml; \
+                fi; \
+                if grep -q "repository_tag" "$dir/values/values.yaml"; then \
+                  sed -i "s/.*repository_tag.*/        repository_tag: v$RELEASE_VERSION/" "$dir"/values/values.yaml; \
+                fi; \
+            fi; \
+          done  
+          # commit and push changes
```
Comment (dvvanessastoiber) [inline .github/workflows/release-source.yml:173]: please remember to add a helm dep update and helm dep build as the last step before running the helm-docs command

File: .github/workflows/release-source.yml (hunk 2 above)
Comment (dv-hossam-radwan) [reply .github/workflows/release-source.yml]: Changes addes

File: .github/workflows/release-source.yml (hunk 3 above)
Comment (dv-hossam-radwan) [reply .github/workflows/release-source.yml]: changes added

File: .github/workflows/release-source.yml (hunk 4 above)
Comment (dv-hossam-radwan) [reply .github/workflows/release-source.yml]: changes added

File: .github/workflows/release-source.yml (hunk 6 above)
Comment (dv-hossam-radwan) [reply .github/workflows/release-source.yml:181]: changes added

**Comment by dv-hossam-radwan** [comment]
Sample output after changes:
https://github.com/datavisyn/test-app/actions/runs/9001672000/job/24728373045.
https://github.com/datavisyn/test-app/pull/57

**Comment by dvvanessastoiber** [comment]
closed in favor of https://github.com/datavisyn/github-workflows/pull/74

**Pull request closed without merging**